hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -426,23 +429,23 @@ public void createEmptyExternalFile(ExecRow execRows, int[] baseColumnMap, int[]
             for (int i = 0; i < partitionBy.length; i++) {
                 partitionByCols.add(ValueRow.getNamedColumn(partitionBy[i]));
             }
-                if (storedAs!=null) {
-                    if (storedAs.toLowerCase().equals("p")) {
-                        empty.write().option("compression",compression).partitionBy(partitionByCols.toArray(new String[partitionByCols.size()]))
-                                .mode(SaveMode.Append).parquet(location);
-                    }
-                    else if (storedAs.toLowerCase().equals("a")) {
-                        empty.write().option("compression",compression).partitionBy(partitionByCols.toArray(new String[partitionByCols.size()]))
-                                .mode(SaveMode.Append).format("com.databricks.spark.avro").save(location);
-                    }
-                    else if (storedAs.toLowerCase().equals("o")) {
-                        empty.write().option("compression",compression).partitionBy(partitionByCols.toArray(new String[partitionByCols.size()]))
-                                .mode(SaveMode.Append).orc(location);
-                    }
-                    if (storedAs.toLowerCase().equals("t")) {
-                        // spark-2.2.0: commons-lang3-3.3.2 does not support 'XXX' timezone, specify 'ZZ' instead
-                        empty.write().option("compression",compression).option("timestampFormat", "yyyy-MM-dd'T'HH:mm:ss.SSSZZ").mode(SaveMode.Append).csv(location);
-                    }
+            if (storedAs!=null) {
+                if (storedAs.toLowerCase().equals("p")) {
+                    empty.write().option("compression",compression).partitionBy(partitionByCols.toArray(new String[partitionByCols.size()]))
+                            .mode(SaveMode.Append).parquet(location);
+                }
+                else if (storedAs.toLowerCase().equals("a")) {
+                    empty.write().partitionBy(partitionByCols.toArray(new String[partitionByCols.size()]))
+                            .mode(SaveMode.Append).format("com.databricks.spark.avro").save(location);
```

Comment:
why option("compression", compression) does not work for avro?

Reply:
I tried but not work.
In the Spark-Avro doc (https://docs.databricks.com/spark/latest/data-sources/read-avro.html), the example showed the compression code was accepted by spark conf settting. 

You can also specify Avro compression options:
Copy to clipboardCopy
import com.databricks.spark.avro._
// configuration to use deflate compression
spark.conf.set("spark.sql.avro.compression.codec", "deflate") 

Others raised the same confusion of compression code setting, but was not responded.
(https://github.com/databricks/spark-avro/issues/259)